jr-annotation-support/src/main/java/com/fasterxml/jackson/jr/annotationsupport/AnnotationBasedValueRWModifier.java

```diff
@@ -75,4 +97,115 @@ public void writeValue(JSONWriter context, JsonGenerator g, Object value) throws
             return _valueType;
         }
     }
+
+    /**
+     * Serialize an enum using the {@link JsonValue} tagged method
+     */
+    private static class EnumJsonValueWriter implements ValueWriter {
+        private final Class<?> _valueType;
+        private final Method _jsonValueMethod;
+
+        private EnumJsonValueWriter(Class<?> _valueType, Method _jsonValueMethod) {
+            this._valueType = _valueType;
+            this._jsonValueMethod = _jsonValueMethod;
+        }
+
+        @Override
+        public void writeValue(JSONWriter context, JsonGenerator g, Object value) throws IOException {
+            try {
+                context.writeValue(_jsonValueMethod.invoke(value));
+            } catch (Exception e) {
+                throw new IOException("Cannot call JsonValue method", e);
+            }
+        }
+
+        @Override
+        public Class<?> valueType() {
+            return _valueType;
+        }
+
+        /**
+         * Scan the methods of the enum to find a {@link JsonValue} tagged method
+         * which takes no parameters and returns a {@link String}
+         * @param type the type
+         * @return either a {@link EnumJsonValueWriter} to write this enum with, or <code>null</code> if no suitable
+         * method found
+         */
+        private static EnumJsonValueWriter of(Class<?> type) {
+            if (!isEnum(type)) {
+                return null;
+            }
+            return getJsonValueFunction(type, type);
+        }
+
+        private static EnumJsonValueWriter getJsonValueFunction(Class<?> type, Class<?> inspectionType) {
+            for (Method method : inspectionType.getDeclaredMethods()) {
+                if (method.getDeclaredAnnotation(JsonValue.class) != null &&
```

Comment:
I would filter out non-static methods to avoid problems (or indicate an error for attempting).

Also not sure what handling should be for non-public methods; jackson-databind can force access but I don't remember if jackson-jr tries that or not.

Reply:
Actually yes, `ValueReader/WriterLocator` do force access as needed. So should be done here too, probably.

Reply:
Done - this might be an issue with illegal reflective access in later Java, though.